addon/mixins/route.js

```diff
@@ -95,10 +105,24 @@ const RouteMixin = Mixin.create({
 
     @method infinityModel
     @param {String} modelName The name of the model.
-    @param {Object} options Optional, the perPage and startingPage to load from.
+    @param {Object} options - optional - the perPage and startingPage to load from.
+    @param {Object} boundParamsOrInfinityModel - optional - 
+      params on route to be looked up on every route request or
+      instance of InfinityModel
     @return {Ember.RSVP.Promise}
   */
-  infinityModel(modelName, options, boundParams) {
+  infinityModel(modelName, options, boundParamsOrInfinityModel) {
+
+    let boundParams, ExtendedInfinityModel;
```

Comment:
So here is what I was initially thinking.  Even though not fully thought through, the 3rd arg may be a class or boundParams.  @hhff is this what you were thinking?

Reply:
Yup!  totally!